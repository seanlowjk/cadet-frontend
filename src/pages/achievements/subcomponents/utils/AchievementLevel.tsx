import React from 'react';
import { ProgressBar } from '@blueprintjs/core';

type AchievementLevelProps = {
  studentExp: number;
};

function AchievementLevel(props: AchievementLevelProps) {
  const { studentExp } = props;

  const expPerLevel = 1000;
  const level = Math.floor(studentExp / expPerLevel);
  const progress = studentExp % expPerLevel;
  const progressFrac = progress / expPerLevel;

  return (
    <>
      <h3>{level}</h3>
      <ProgressBar className="progress" value={progressFrac} />
    </>
  );
}

export default AchievementLevel;
