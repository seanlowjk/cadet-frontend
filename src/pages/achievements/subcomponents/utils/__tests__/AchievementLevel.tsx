import { mount } from 'enzyme';
import React from 'react';

import AchievementLevel from '../AchievementLevel';

const mockProps = {
  studentExp: 0
};

test('AchievementExp component renders correctly', () => {
  const app = <AchievementLevel {...mockProps} />;
  const tree = mount(app);
  expect(tree.debug()).toMatchSnapshot();
});
