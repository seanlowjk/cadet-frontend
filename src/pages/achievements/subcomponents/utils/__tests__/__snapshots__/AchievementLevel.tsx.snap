// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AchievementExp component renders correctly 1`] = `
"<AchievementLevel studentExp={0}>
  <h3>
    0
  </h3>
  <Blueprint3.ProgressBar className=\\"level-progress\\" value={0}>
    <div className=\\"bp3-progress-bar level-progress\\">
      <div className=\\"bp3-progress-meter\\" style={{...}} />
    </div>
  </Blueprint3.ProgressBar>
</AchievementLevel>"
`;
